cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -985,16 +988,25 @@ func shouldDeleteNode(ctx *controllerContext, node *v1.Node, getInstance func(*c
 }
 
 func deleteAllPodsBoundToNode(ctx *controllerContext, nodeName string) error {
-	errs := []error{}
 	deletedPods := []string{}
 
-	podList, err := ctx.client.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{FieldSelector: "spec.nodeName=" + nodeName})
+	podList, err := ctx.client.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{FieldSelector: "spec.nodeName=" + nodeName, ResourceVersion: "0"})
 	if err != nil {
-		errs = append(errs, err)
+		return fmt.Errorf("failed to list pods on node: %v: %v", nodeName, err)
 	}
 
+	errs := []error{}
 	for _, pod := range podList.Items {
-		err := ctx.client.CoreV1().Pods(pod.Namespace).Delete(context.TODO(), pod.Name, *metav1.NewDeleteOptions(0))
+		pod := pod
```

Comment:
nit: this doesn't look necessary (we don't store a pointer to this variable)

Reply:
(removed)